builtin/client/register.lua

```diff
@@ -24,23 +27,30 @@ function core.run_callbacks(callbacks, mode, ...)
 	for i = 1, cb_len do
 		local cb_ret = callbacks[i](...)
 
-		if mode == 0 and i == 1 or mode == 1 and i == cb_len then
+		if mode == 0 and i == 1 then
+			ret = cb_ret
+		elseif mode == 1 and i == cb_len then
```

Comment:
>  Other modes also differed a bit from spec

Good catch, but please fix them in a separate PR because it needs to be tested differently. Changing `RUN_CALLBACKS_MODE_AND_SC` is okay. This mode is not used anywhere else.

Reply:
Ok.
I've moved the change to here: https://github.com/Desour/minetest/tree/fix_run_callbacks
I won't open a new PR though, as that would be a waste of time.